nmigen/hdl/ast.py

```diff
@@ -834,6 +834,11 @@ def __init__(self, shape=None, *, name=None, reset=0, reset_less=False, min=None
                  attrs=None, decoder=None, src_loc_at=0):
         super().__init__(src_loc_at=src_loc_at)
 
+        if isinstance(reset, Enum):
+            reset = reset.value
+        if not isinstance(reset, int):
+            raise ValueError("Reset has to be an int or an integral Enum")
```

Comment:
This should be a `TypeError`, and talk about reset *value*, not just reset, which is a different concept.

Reply:
Done.